pkg/monitor/process.go

```diff
@@ -151,3 +153,88 @@ func (p *Process) stop() {
 	p.state = stopped
 	p.cmd = nil
 }
+
+func (p *Process) killAllByName() {
+	p.logger.Infof("killing process %s", p.Name)
+
+	procList, err := process.Processes()
+	if err != nil {
+		p.logger.WithError(err).Error("failed to call process.Processes")
+		return
+	}
+
+	for _, proc := range procList {
+		if proc.Pid == 0 {
+			continue
+		}
+
+		name, err := proc.Name()
+		if err != nil {
+			// it's expected go get EOF if the process no longer exists at this point.
+			if err != io.EOF {
+				log.Infof("error getting name for process %d: %s", proc.Pid, err.Error())
+			}
+			continue
+		}
+
+		if name == "" {
+			continue
+		}
+
+		if !strings.Contains(name, p.Name) {
+			continue
+		}
+
+		log.Infof("terminating process %s with pid %d...", p.Name, proc.Pid)
+		if err := terminate(proc); err != nil {
+			log.Infof("error terminating process %s with pid %d: %s", p.Name, proc.Pid, err.Error())
+		}
+		log.Infof("terminated process %s with pid %d", proc.Pid)
+	}
+
+	p.logger.Infof("process %s killed", p.Name)
+}
+
+func terminate(p *process.Process) error {
+	if err := p.Terminate(); err != nil {
+		return err
+	}
+
+	notRunning, err := waitUntilNotRunning(p)
+	if err != nil {
+		return err
+	}
+
+	if notRunning {
+		return nil
+	}
+
+	if err := p.Kill(); err != nil {
```

Comment:
if `notRunning` is `false`, why do we kill the process?

Reply:
`notRunning` to false means it is running, no?

Reply:
ah right, it's a doble negation. could we instead switch to 'isRunning' so the condition is easier to read?